### ISSUE TITLE
Application lifecycle

### ISSUE THREAD

**Issue by dmitry-korolev** [TinkoffCreditSystems/stapp#1]
Problem and motivation:
Currently, an application starts when created. That forces the developer to create wrappers to delay the initialization. Also, there is no way to "destroy" an application (force unsubscribe all listeners and let the GC do its work). Finally, "dependencies" mechanism looks utterly useless.

Solution:
* Remove `dependencies` field from `createApp` config.
* Introduce two methods to the application: `init` and `destroy`.
* `init` method should initialize modules with provided dependencies, start an engine and dispatch a special `init` event.
* Application state still should be subscribable at any time.
* `destroy` method should unsubscribe all store subscriptions and dispatch a special `destroy` event.

**Comment by dmitry-korolev** [comment]
examples:
```typescript
const app = createApp({
	name: 'test',
	modules: [...]
})

const { Consumer } = createComponents(app)

// later
app.init(dependencies)

// later
app.destroy()
```

**Comment by dmitry-korolev** [comment]
Should app be subscribable before init? If not, should `.init` return a new app each time?

**Comment by dmitry-korolev** [comment]
Destroy
1. Emits `destroy` event
2. Completes event$ and state$ streams
3. Unsubscribes from epics
4. Unsubscribes devtools

**Comment by dmitry-korolev** [comment]
closed in #87
